Geometry/HGCalCommonData/src/HGCalParametersFromDD.cc

```diff
@@ -52,11 +59,34 @@ bool HGCalParametersFromDD::build(const DDCompactView* cpv,
   DDSpecificsMatchesValueFilter filter{val};
   DDFilteredView fv(*cpv,filter);
   bool ok = fv.firstChild();
+  int  mode(0);
 
   if (ok) {
     DDsvalues_type sv(fv.mergedSpecifics());
-    php.mode_ = getGeometryMode("GeometryMode", sv);
+    php.mode_ = getGeometryMode("GeometryMode", sv, true);
```

Comment:
indeed, perhaps I am. however, can mode_ in HGCalParameters and the return type of getGeometryMode() not be updated to return the enum type. That would save a lot of casting to and from ints for no reason.